app/src/main/java/com/example/tasky/agenda_details/presentation/AgendaDetailsViewModel.kt

```diff
@@ -145,12 +146,101 @@ class AgendaDetailsViewModel @Inject constructor(
         //set viewstate to editing
     }
 
-    fun delete() {
+    fun delete(agendaItemType: AgendaItemType) {
         //todo delete functionality; navigate back to agenda
     }
 
-    fun save() {
+    fun save(agendaItemType: AgendaItemType) {
```

Comment:
Isn't this an attribute you have available globally in this ViewModel? In that case, you wouldn't need to pass it as a parameter. Should just be a navigation argument, which you could retrieve using SavedStateHandle:
```
@HiltViewModel
class AgendaDetailsViewModel @Inject constructor(
    private val savedStateHandle: SavedStateHandle,
    ...
): ViewModel() {

    // Assuming agendaItemType is passed as a string navigation argument
    private val agendaItemType = savedStateHandle.get<String>("agendaItemType")?.let {
        AgendaItemType.valueOf(it)
    }
}
```